ompi/mca/coll/libnbc/nbc.c

```diff
@@ -692,39 +645,93 @@ int  NBC_Init_comm(MPI_Comm comm, NBC_Comminfo *comminfo) {
   return OMPI_SUCCESS;
 }
 
-int NBC_Start(NBC_Handle *handle, NBC_Schedule *schedule) {
+int NBC_Start(NBC_Handle *handle) {
   int res;
 
-  handle->schedule = schedule;
-
+  /* bozo case */
+  if ((ompi_request_t *)handle == &ompi_request_empty) {
+    return OMPI_SUCCESS;
+  }
   /* kick off first round */
   res = NBC_Start_round(handle);
   if (OPAL_UNLIKELY(OMPI_SUCCESS != res)) {
     return res;
   }
+  handle->super.req_state = OMPI_REQUEST_ACTIVE;
```

Comment:
@dholmes-epcc-ed-ac-uk Thanks. Regarding `ACTIVE`, I'll update the PR next week. Regarding `nbc_complete`, the commit message 4866c82 explains the necessity. I need the status information before `req_state` or `req_complete` is updated because `NBC_Progress` is called recursively. I know it is not a smart solution.